odoo/addons/test_new_api/tests/test_new_fields.py

```diff
@@ -3124,6 +3124,57 @@ def test_unlink_asbtract(self):
         ], limit=1).unlink()
 
 
+@common.tagged('selection_update_base')
+class TestSelectionUpdates(common.TransactionCase):
+    MODEL_BASE = 'test_new_api.model_selection_base'
+    MODEL_RELATED = 'test_new_api.model_selection_related'
+    MODEL_RELATED_UPDATE = 'test_new_api.model_selection_related_updatable'
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        # Specifying a lang in env/context should not increase query counts
+        # of CRUD operations
+        cls.env = cls.env(context={'lang': 'en-US'})
+
+    def test_selection_create(self):
+        self.env.cr.now()  # Do not consider in query counts
+        with self.assertQueryCount(2):  # defaults + INSERT
+            self.env[self.MODEL_BASE].create({})
+        with self.assertQueryCount(1):
+            self.env[self.MODEL_BASE].create({'my_selection': 'foo'})
+
+    def test_selection_update(self):
+        record = self.env[self.MODEL_BASE].create({'my_selection': 'foo'})
+        with self.assertQueryCount(3):  # SELECT, SELECT (related field), UPDATE
+            record.my_selection = 'bar'
+            record.flush()
+
+    def test_selection_related_create(self):
+        related_record = self.env[self.MODEL_BASE].create({'my_selection': 'foo'})
+        with self.assertQueryCount(2):  # defaults (readonly related field), INSERT
+            self.env[self.MODEL_RELATED].create({'res_id': related_record.id})
+
+    def test_selection_related_noupdate_update(self):
+        related_record = self.env[self.MODEL_BASE].create({'my_selection': 'foo'})
+        record = self.env[self.MODEL_RELATED].create({'res_id': related_record.id})
+        with self.assertQueryCount(0):  # discarded diff bc readonly related
+            record.related_selection = 'bar'
+            record.flush()
+
+    def test_selection_related_update_create(self):
+        related_record = self.env[self.MODEL_BASE].create({'my_selection': 'foo'})
+        with self.assertQueryCount(2):  # defaults (related field), INSERT
+            self.env[self.MODEL_RELATED_UPDATE].create({'res_id': related_record.id})
+
+    def test_selection_related_update_update(self):
+        related_record = self.env[self.MODEL_BASE].create({'my_selection': 'foo'})
+        record = self.env[self.MODEL_RELATED_UPDATE].create({'res_id': related_record.id})
+        with self.assertQueryCount(3):  # SELECT (related recs), SELECT (related recs), UPDATE
+            record.related_selection = 'bar'
+            record.flush()
```

Comment:
You could merge both methods: one query count for creating the record, one query count for updating it.